django_netjsonconfig/base/template.py

```diff
@@ -99,6 +137,15 @@ def clean(self, *args, **kwargs):
             self.auto_cert = False
         if self.type == 'vpn' and not self.config:
             self.config = self.vpn.auto_client(auto_cert=self.auto_cert)
+        if self.flag == 'public' or self.flag == 'shared_secret':
+            if not self.description:
+                raise ValidationError({'description': _('Please enter public description of '
+                                                        'shared template')})
+            if not self.notes:
+                raise ValidationError({'notes': _('Please enter notes used by administrations of '
+                                                  'shared template')})
+            if self.variable == {}:
```

Comment:
this too, I would not make it required for now, there may be some standard templates without variables, we don't know now